docs/core/extensions/httpclient-factory-troubleshooting.md

```diff
@@ -0,0 +1,184 @@
+
+## `HttpClient` doesn't respect `Scoped` lifetime
+
+You can hit a problem if you need to access any scoped service, for example, `HttpContext`, or some scoped cache, from within the `HttpMessageHandler`. The data saved there can either "disappear", or, the other way around, "persist" when it should not. This is caused by the DI **scope mismatch** between the application context and the handler instance, and it is a known limitation in `IHttpClientFactory`.
+
+`IHttpClientFactory` creates a separate DI scope per each `HttpMessageHandler` instance. These handler scopes are distinct from application context scopes (for example, ASP.NET incoming request scope, or a user-created manual DI scope), so they will not share scoped service instances.
+
+It means that:
+
+- Any data cached "externally" in a scoped service will NOT be available within the `HttpMessageHandler`
+- Any data cached "internally" within the `HttpMessageHandler` or its scoped dependencies CAN be observed from multiple application DI scopes (for example, from different incoming requests) as they can share the same handler.
+
+❌ DO NOT cache any scope-related information (such as data from `HttpContext`) inside `HttpMessageHandler` instances or its dependencies to avoid leaking sensitive information.
+
+❌ DO NOT use cookies, as the `CookieContainer` will be shared together with the handler.
+
+✔️ CONSIDER not storing the information, or only pass it within the `HttpRequestMessage` instance.
+
+To pass arbitrary information alongside the `HttpRequestMessage`, you can use the <xref:System.Net.Http.HttpRequestMessage.Options?displayProperty=nameWithType> property.
+
+✔️ CONSIDER incapsulating all the scope-related (for example, authentication) logic in a separate `DelegatingHandler`, that is NOT created by the `IHttpClientFactory`, and use it to wrap the `IHttpClientFactory`-created handler.
+
+To create just an `HttpMessageHandler` without `HttpClient`, call <xref:System.Net.Http.IHttpMessageHandlerFactory.CreateHandler%2A?displayProperty=nameWithType> for any registered _named client_. In that case, you will need to create an `HttpClient` instance yourself using the combined handler. You can find a fully runnable example for this workaround on [GitHub](https://github.com/dotnet/docs/tree/main/docs/core/extensions/snippets/http/scopeworkaround).
+
+For more information, see the [Message Handler Scopes in IHttpClientFactory](httpclient-factory.md#message-handler-scopes-in-ihttpclientfactory) section in the `IHttpClientFactory` guidelines.
+
+## `HttpClient` doesn't respect DNS changes
+
+Even if `IHttpClientFactory` is used, it is still possible to hit the stale DNS problem. This can usually happen if an `HttpClient` instance gets captured in a `Singleton` service, or, in general, stored somewhere for the time longer than the specified `HandlerLifetime`. `HttpClient` will also get captured if the respective _typed client_ gets captured by a singleton.
```

Comment:
```suggestion
Even if `IHttpClientFactory` is used, it is still possible to hit the stale DNS problem. This can usually happen if an `HttpClient` instance gets captured in a `Singleton` service, or, in general, stored somewhere for a period of time that's longer than the specified `HandlerLifetime`. `HttpClient` will also get captured if the respective _typed client_ is captured by a singleton.
```